### PR TITLE
fix(repl): Accept tab when previous character is space

### PR DESCRIPTION
**Pull request by sigmaSd**
This adds a heuristic that allows tabbing, if the user hits tab and the previous character on the line is white space he's most likely asking for an actual tab not completion.
Python repl for example does this.

https://user-images.githubusercontent.com/22427111/174358649-64c58ee3-5670-4583-ba44-44ceb31ac6e0.mov

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Seems like a great usability addition; is it possible to add a test for this feature?

@dsherret please take a look too

File: cli/tools/repl/editor.rs [hunk 1]
```diff
@@ -391,3 +396,28 @@ impl ReplEditor {
     Ok(())
   }
 }
+
+struct TabEventHandler;
+impl ConditionalEventHandler for TabEventHandler {
```
[the remark this answers is not in the document]
Comment (sigmaSd) [reply cli/tools/repl/editor.rs:406]: done

File: cli/tools/repl/editor.rs [hunk 2]
```diff
@@ -391,3 +396,33 @@ impl ReplEditor {
...
+  fn handle(
+    &self,
+    evt: &Event,
+    n: RepeatCount,
+    _: bool,
+    ctx: &EventContext,
+  ) -> Option<Cmd> {
+    debug_assert_eq!(*evt, Event::from(KeyEvent::from('\t')));
+    if ctx.line().is_empty()
+      || ctx.line()[..ctx.pos()]
```
Comment (dsherret) [inline cli/tools/repl/editor.rs:416]: To save other reviewers time, `ctx.pos()` is indeed byte position https://docs.rs/rustyline/9.1.2/rustyline/struct.EventContext.html

**Review by dsherret — COMMENTED** [review]
Thanks for this @sigmaSd! There are some pty tests for the repl here: https://github.com/denoland/deno/blob/main/cli/tests/integration/repl_tests.rs -- Would you be able to check if it's possible to add one for this?

**Comment by sigmaSd** [comment]
I added a simple test, do you have some ideas to improve it?

**Review by dsherret — APPROVED** [review]
LGTM. Thanks @sigmaSd!

**Pull request merged**